### PR TITLE
Periodically reload allowlist from file

### PR DESCRIPTION
**Pull request by ian-shim**
## Why are these changes needed?
Deprecated `auth.allowlist` and periodically reloads allowlist from the json file specified in `auth.allowlist-file` 

- [ ]  Merge only when `auth.allowlist` is no longer in use

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/apiserver/server.go [hunk 1]
```diff
@@ -778,6 +795,15 @@ func (s *DispersalServer) Start(ctx context.Context) error {
 	return nil
 }
 
+func (s *DispersalServer) LoadAllowlist() {
+	al, err := ReadAllowlistFromFile(s.rateConfig.AllowlistFile)
+	if err != nil {
+		s.logger.Error("failed to load allowlist", "err", err)
```
Comment (dmanc) [inline disperser/apiserver/server.go:801]: So if it fails to load allowlist, it will just keep the old values?

Comment (ian-shim) [reply disperser/apiserver/server.go:801]: Yes!

File: disperser/apiserver/server.go [hunk 2]
```diff
@@ -778,6 +795,15 @@ func (s *DispersalServer) Start(ctx context.Context) error {
...
+		return
+	}
+	s.rateConfig.Allowlist = al
```
Comment (mooselumph) [inline disperser/apiserver/server.go:804]: Does this need any locks?

**Review by mooselumph — APPROVED** [review]

File: disperser/apiserver/server.go (hunk 2 above)
Comment (ian-shim) [reply disperser/apiserver/server.go:804]: I don't think this really needs to be synchronized. 
The worst case is that a potentially disallowed request goes through at the moment new allowlist is loaded (or vice versa)

**Pull request merged**